langchain/src/chat_models/anthropic.ts

```diff
@@ -295,47 +278,37 @@ export class ChatAnthropic extends BaseChatModel implements AnthropicInput {
         });
       }
       makeCompletionRequest = async () => {
-        try {
-          const stream = this.streamingClient.completions.create({
-            ...request,
-          });
-
-          stream
-            .then(async (stream) => {
-              for await (const data of stream) {
-                if (options.signal?.aborted) {
-                  stream.controller.abort();
-                }
-
-                if (data.stop_reason) {
-                  break;
-                }
-                const part = data.completion;
-                if (part) {
-                  // eslint-disable-next-line no-void
-                  void runManager?.handleLLMNewToken(part ?? "");
-                }
-              }
-            })
-            .catch((e: any) => {
-              // Anthropic doesn't actually throw JavaScript error objects at the moment.
-              // We convert the error so the async caller can recognize it correctly.
-              if (e?.name === "AbortError") {
-                throw new Error(`${e.name}: ${e.message}`);
-              }
-              throw e;
-            });
-
-          return stream;
-          // eslint-disable-next-line @typescript-eslint/no-explicit-any
-        } catch (e: any) {
-          // Anthropic doesn't actually throw JavaScript error objects at the moment.
-          // We convert the error so the async caller can recognize it correctly.
-          if (e?.name === "AbortError") {
-            throw new Error(`${e.name}: ${e.message}`);
+        const stream = await this.streamingClient.completions.create({
+          ...request,
+        });
+
+        const completion: AnthropicApi.Completion = {
+          completion: "",
+          model: "",
+          stop_reason: "",
+        };
+
+        for await (const data of stream) {
+          completion.stop_reason = data.stop_reason;
+          completion.model = data.model;
+
+          if (options.signal?.aborted) {
+            stream.controller.abort();
+            throw new Error("AbortError: User aborted the request.");
```

Comment:
FYI, you _might_ want this to be more like:

```ts
const err = new Error("User aborted the request.")
err.name = "AbortError";
throw err;
```
which I believe to be closer to what JS runtimes do.

Reply:
This gets caught by our caller class which would catch this - we could/should change it in the future but it's out of scope for this one.